### ISSUE TITLE
Themes for white label (first steps)

### ISSUE THREAD

**Issue by ackernaut** [ParabolInc/action#1839]
## Enhancement

When other orgs want to use our software with clients, we need a minimal system for theming so that we can provision and maintain their instances quickly and easily.

### Acceptance Criteria
- The theming system maps across contextual affordances for both branding colors (eg. sidebar is branded) and UI colors (eg. green dot for connected).
- The theming system abstracts and surfaces a simple file where most of the magic can be updated as theme configuration.
- The theming supports brand logos
- This work involves brand/palette/theme constants that style the app.

#### In the future:
- Consider adding a view to update the values of the theme in a SCRAPPY admin.
- Also, a view or sub-view to update the org logo for the theme.
- Also, consider exposing this admin view to orgs for their designers to update.

#### For reference:
[https://github.com/emotion-js/emotion/tree/master/packages/emotion-theming](https://github.com/emotion-js/emotion/tree/master/packages/emotion-theming)

Also, see doc in Drive for input from our partner

### Estimated Effort
- 12 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by ackernaut** [comment]
Added a note on using [emotion-theming](https://github.com/emotion-js/emotion/tree/master/packages/emotion-theming)

**Comment by ackernaut** [comment]
This is largely coming from the solution space. Once we come up with a way forward we can optionally create an implementation issue.

**Comment by ackernaut** [comment]
Moving this back to the backlog as #1889 needs to come first.

**Comment by mattkrick** [comment]
we could do all the theme stuff at build time. Here's how it'd look:

1. Refactor the ui.js/theme.js, etc. stuff so there are 2 outputs: a json file, and a file of functions that use that json file. i think this was the original idea years ago, but we've strayed. call it THEME.json, and UI.js

2. For each white label domain, build a custom json file to be served. call it WHITE_LABEL_THEME.json

3. When a request comes from a white label referrer, include the correct WHITE_LABEL_THEME.json in the bundle. A quick & dirty way is to set it to window.__THEME__ in the html that gets sent down

4. on the client, merge the 2 json files, since THEME.json is a singleton, all references stay intact & the styles will update automatically


This has the advantage of being faster to ship & better performance.

NOTE: if we did SSR, this would cause a fouc. currently, we only SSR the login page, so it's not a huge deal. If, in the future, we used cookies & did SSR for all our pages, then we'd want to do separate builds for each white label. None of the social networks today SSR your private data (eg newsfeed), so i don't see why we should. They probably don't because then they can serve a single logged-in landing page via a cdn & you see something faster, but that's just a guess.
